pkg/storage/snapshot/restore.go

```diff
@@ -1403,6 +1410,19 @@ func (ctrl *VMRestoreController) volumesNotForRestore(content *snapshotv1.Virtua
 		}
 	}
 
+	if !target.Exists() {
+		// Restoring backend storage not supported when target is not equal to the source
+		volumes, err := storageutils.GetVolumes(content.Spec.Source.VirtualMachine, ctrl.Client, storageutils.WithBackendVolume)
+		if err != nil && !storageutils.IsErrNoBackendPVC(err) {
+			return noRestore, err
+		}
+		for _, volume := range volumes {
```

Comment:
why not unit all this code with the previous filtering? i.e do the getVolumes of all the volumes including the backendvolume at the beginning and in the err check and in the for loop if target exist filter also the backend pvc..